cinema/views.py

```diff
@@ -45,14 +69,61 @@
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = (
+        MovieSession.objects
+        .select_related("movie", "cinema_hall")
+        .annotate(
+            tickets_available=(
+                F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
+        )
+    )
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+        return queryset.distinct()
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
-
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
-
         return MovieSessionSerializer
+
+
+class OrderResultsSetPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = "page_size"
+    max_page_size = 10
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.prefetch_related(
+        "tickets__movie_session__movie",
+        "tickets__movie_session__cinema_hall")
+    serializer_class = OrderSerializer
+    pagination_class = OrderResultsSetPagination
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
```

Comment:
Filtering the queryset by `user=self.request.user` is a good practice to ensure users can only access their own orders. Ensure that the `Order` model has a `user` field and that it is correctly set up to relate to the user model.